### PR TITLE
K8s integration test with different k8s versions

### LINKED ISSUE #2350: Improve K8s integration test coverage
In terms of K8s test coverage, we currently test against, and a have a client dependency set to our lowest supported version (1.18). The client dependency prevents us from making calls that aren't supported by the 1.18 APIs but doesn't actually give us assurance that the code still works on the latest K8s releases.

Further, since our test coverage only covers our minimum version, we wouldn't discover any such breaks even with integration testing.

What we really need is a series of integration tests that can run against all supported kubernetes versions, or at a minimum, the lowest and the greatest. This would give us flexibility to change our client deps and reap those benefits (better interfaces, code fixes, etc) without sacrificing compatibility guarantees. The hidden requirement would be, of course, that any new code that invokes kubernetes APIs is also covered by the integration tests.

### PR DESCRIPTION
**Pull request by FedeNQ**
**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
Integration test (k8s)

**Description of change**
Fetch the lastest versions (from 1.21 to latest) of k8s deployed on docker.hub, and use them to run the k8s integration test.
Allow to add specific configurations of k8s and kind version.

**Which issue this PR fixes**
Fixes #2350

### CODE REVIEW

File: test/integration/suites-k8s/k8s/00-setup [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (amartinezfayo) [inline test/integration/suites-k8s/k8s/00-setup:1]: I'm not sure if there is a good reason to have this in a completely separated suite. I would probably try to have this as part of the existing k8s integration test.

**Review by amartinezfayo — COMMENTED** [review]
Thank you @FedeNQ for this contribution, it's something that we really needed!
It's looking great. I left some comments.

I think that the main thing could be having this as part of the existing test vs duplicating it with the new functionality.

**Review by amartinezfayo — COMMENTED** [review]
Thank you @FedeNQ for addressing the comments.
Looking at the PR build, it seems that it ran the admin-endpoints test instead of the k8s test? E.g.: https://github.com/spiffe/spire/actions/runs/5881025153/job/15988654799#step:10:14
I may be missing something, but looks like it did that for all the steps for the k8s tests. Could you please look at that?

I have a couple of additional comments:
- I'm realizing that we should also make this part of the release build workflow, please add this it also.
- I'm thinking that the .txt files could just be in the suites/k8s folder. What do you think?

**Review by amartinezfayo — APPROVED** [review]
Thank you @FedeNQ@
